### ISSUE TITLE
WildflyIT.testTransportClient failed in CI

### ISSUE THREAD

**Issue by imotov** [elastic/elasticsearch#24436]
Failed in CI today - https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=amazon/911/console
```
  2> NOTE: Linux 4.4.44-39.55.amzn1.x86_64 amd64/Oracle Corporation 1.8.0_121 (64-bit)/cpus=4,threads=1,free=483862184,total=514850816
  2> NOTE: All tests run in this JVM: [WildflyIT]
   > Expected: <201>
   >      but: was <400>
   > 	at __randomizedtesting.SeedInfo.seed([990E732BDE24D8EF:F5B837CA4E02164A]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.wildfly.WildflyIT.testTransportClient(WildflyIT.java:78)
   > 	at java.lang.Thread.run(Thread.java:745)
Completed [1/1] in 1.11s, 1 test, 1 failure <<< FAILURES!

[ant:junit4] JVM J0: stderr was not empty, see: /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob-unix-compatibility/os/amazon/qa/wildfly/build/testrun/integTestRunner/temp/junit4-J0-20170502_094849_4827938408300352232738.syserr
Tests with failures:
  - org.elasticsearch.wildfly.WildflyIT.testTransportClient
```

It seems to be the first failure with this particular error. However this test failed 6 days ago with a different, but possibly related error:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+multijob-intake/1073/console

```
2> NOTE: reproduce with: ant test  -Dtestcase=WildflyIT -Dtests.method=testTransportClient -Dtests.seed=88966EDD55EC9F58 -Dtests.locale=hr -Dtests.timezone=America/St_Kitts -Dtests.asserts=true -Dtests.file.encoding=UTF8
ERROR   0.39s | WildflyIT.testTransportClient <<< FAILURES!
   > Throwable #1: org.apache.http.client.ClientProtocolException
  2> NOTE: test params are: codec=Asserting(Lucene62): {}, docValues:{}, maxPointsInLeafNode=677, maxMBSortInHeap=6.504225533270213, sim=RandomSimilarity(queryNorm=false,coord=yes): {}, locale=hr, timezone=America/St_Kitts
  2> NOTE: Linux 4.4.44-39.55.amzn1.x86_64 amd64/Oracle Corporation 1.8.0_121 (64-bit)/cpus=4,threads=1,free=487412120,total=514850816
  2> NOTE: All tests run in this JVM: [WildflyIT]
   > 	at __randomizedtesting.SeedInfo.seed([88966EDD55EC9F58:E4202A3CC5CA51FD]:0)
   > 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:186)
   > 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
   > 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
   > 	at org.elasticsearch.wildfly.WildflyIT.testTransportClient(WildflyIT.java:77)
   > 	at java.lang.Thread.run(Thread.java:745)
   > Caused by: org.apache.http.ProtocolException: The server failed to respond with a valid HTTP response
   > 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:151)
   > 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57)
   > 	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
   > 	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
   > 	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:167)
   > 	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
   > 	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
   > 	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:271)
   > 	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
   > 	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
   > 	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
   > 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
   > 	... 39 more
Completed [1/1] in 1.09s, 1 test, 1 error <<< FAILURES!

```

**Comment by jasontedor** [comment]
From the logs, HTTP services did not fully startup:

```
2017-05-02 09:48:43,993 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-2) MSC000001: Failed to start service jboss.undertow.listener.default: org.jboss.msc.service.StartException in service jboss.undertow.listener.default: Could not start http listener
	at org.wildfly.extension.undertow.ListenerService.start(ListenerService.java:142)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1948)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1881)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.BindException: Address already in use
	at sun.nio.ch.Net.bind0(Native Method)
	at sun.nio.ch.Net.bind(Net.java:433)
	at sun.nio.ch.Net.bind(Net.java:425)
	at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)
	at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
	at org.xnio.nio.NioXnioWorker.createTcpConnectionServer(NioXnioWorker.java:190)
	at org.xnio.XnioWorker.createStreamConnectionServer(XnioWorker.java:243)
	at org.wildfly.extension.undertow.HttpListenerService.startListening(HttpListenerService.java:126)
	at org.wildfly.extension.undertow.ListenerService.start(ListenerService.java:138)
	... 5 more
```

This is a known issue that can happen with the Wildfly tests because right now we can not use ephemeral ports. I have opened a PR against Wildfly (wildfly/wildfly#9943) so that we can move to ephemeral ports but there's a follow-up PR needed against another Wildfly component before we can fully implement this (we need changes on our side too). This is very low on my priority list.

**Comment by jasontedor** [comment]
To be clear, both build failures referenced here have the same cause.
